cluster-api/src/main/java/io/aiven/klaw/clusterapi/controller/TopicContentsController.java

```diff
@@ -23,7 +23,7 @@ public class TopicContentsController {
       value =
           "/getTopicContents/{bootstrapServers}/"
               + "{protocol}/{consumerGroupId}/{topicName}/{offsetPosition}/partitionId/{selectedPartitionId}/"
-              + "selectedNumberOfOffsets/{selectedNumberOfOffsets}/{clusterIdentification}",
+              + "selectedNumberOfOffsets/{selectedNumberOfOffsets}/{clusterIdentification}/rangeOffsets/{rangeOffsetsStart}/{rangeOffsetsEnd}",
```

Comment:
can offset position be custom or range or from FE ? don't introduce this action in url ?

Reply:
When I work on FE I will make appropriate changes so that the FE will give `offsetPosition` as the two existing ones we have (number and custom) plus it will also give range, if thats what the user selects. I preferred to keep this PR only for BE first, then I can work on FE and make another PR.

As far as changes in url, I was a little confused as to how to approach the problem. In current behavior the `offsetPosition ` is used in two cases

1. `offsetPosition `can be a number. in this case we ignore the `selectedNumberOfOffsets` and `selectedPartitionId`  value in the url. Using the `offsetPosition` number we seek to beginning of topic and pull all data for all partitions upto the given `offsetPosition` number.
2. `offsetPosition` is the string value **custom** in this case we use the provided values of `selectedNumberOfOffsets` and `selectedPartitionId` to only pull data from the provided partition upto the the number provided in `selectedNumberOfOffsets`

Therefore in my case I felt I had the following options

1. Reuse the existing url and try to do the range, in this case I have to figure out how to reuse `offsetPosition`, `selectedNumberOfOffsets` and `selectedPartitionId`. If `offsetPosition` is used to provide the value **range** then that leaves me with the other two for start and end of range, therefore the data will be for ALL partitions.
2. Make some changes to url to accommodate the range.

Would you prefer that we keep the existing url exactly same?

Reply:
How about offsetPosition = { number, custom, range }
Existing situation, if it's number, or custom, nothing changes in the url. 

If it's range, we need partitionId, start and end offsets.
Looks good the way you have done it.
"/getTopicContents/{bootstrapServers}/"
              + "{protocol}/{consumerGroupId}/{topicName}/{offsetPosition}/partitionId/{selectedPartitionId}/"
              + "selectedNumberOfOffsets/{selectedNumberOfOffsets}/{clusterIdentification}/"
              + "rangeOffsets/{rangeOffsetsStart}/{rangeOffsetsEnd}",